design_drafts/components_architecture_and_urdf_examples.md

```diff
@@ -415,285 +414,284 @@ Note:
-      <commandInterfaceType>effort_limit</commandInterfaceType>
-      <stateInterfaceType>position</stateInterfaceType>
-      <stateInterfaceType>velocity</stateInterfaceType>
-      <stateInterfaceType>effort</stateInterfaceType>
-      <stateInterfaceType>coil_resistance</stateInterfaceType>
-      <stateInterfaceType>timestamp</stateInterfaceType>
+      </command_interface>
+      <command_interface>Kp</command_interface>
+      <command_interface>Kd</command_interface>
+      <command_interface>effort_limit</command_interface>
+      <state_interface>position</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>effort</state_interface>
+      <state_interface>coil_resistance</state_interface>
+      <state_interface>timestamp</state_interface>
     </joint>
     <joint name="hind_right_leg_joint3">
-      <commandInterfaceType name="position">
+      <command_interface name="position">
         <param name="min">-1</param>
         <param name="max">1</param>
-      </commandInterfaceType>
-      <commandInterfaceType name="velocity">
+      </command_interface>
+      <command_interface name="velocity">
         <param name="min">-1</param>
         <param name="max">1</param>
-      </commandInterfaceType>
-      <commandInterfaceType name="effort">
+      </command_interface>
+      <command_interface name="effort">
         <param name="min">-0.5</param>
         <param name="max">0.5</param>
-      </commandInterfaceType>
-      <commandInterfaceType>Kp</commandInterfaceType>
-      <commandInterfaceType>Kd</commandInterfaceType>
-      <commandInterfaceType>effort_limit</commandInterfaceType>
-      <stateInterfaceType>position</stateInterfaceType>
-      <stateInterfaceType>velocity</stateInterfaceType>
-      <stateInterfaceType>effort</stateInterfaceType>
-      <stateInterfaceType>coil_resistance</stateInterfaceType>
-      <stateInterfaceType>contact_information</stateInterfaceType>
-      <stateInterfaceType>timestamp</stateInterfaceType>
+      </command_interface>
+      <command_interface>Kp</command_interface>
+      <command_interface>Kd</command_interface>
+      <command_interface>effort_limit</command_interface>
+      <state_interface>position</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>effort</state_interface>
+      <state_interface>coil_resistance</state_interface>
+      <state_interface>contact_information</state_interface>
+      <state_interface>timestamp</state_interface>
     </joint>
 
     <sensor name="sensor1">
       <classType>ros2_control_components/IMUSensor</classType>
-      <stateInterfaceType>orientation</stateInterfaceType>
-      <stateInterfaceType>velocity</stateInterfaceType>
-      <stateInterfaceType>acceleration</stateInterfaceType>
+      <state_interface>orientation</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>acceleration</state_interface>
```

Comment:
```suggestion
      <state_interface name="orientation">
          <param name="min">-54</param>
          <param name="max">23</param>
      </state_interface>
      <state_interface name="velocity"/>
      <state_interface name="acceleration">
          <param name="min">-10</param>
          <param name="max">10</param>
      </state_interface>
```

Reply:
limits on state interfaces is a discussion we need to have in the new year

Reply:
I agree.

The long version:

we should leave them as optional arguments. I would add it for now, and what users are doing with it, is their decision. We don't have to provide any handling for them now.

The reason behind it, is that sometimes is good to know the limits of your sensor to know if the data are reliable. E.g. every FTS has calibration range, but the sensors itself can provide data also outside this range (or should this be sensor-specific parameters)

Reply:
:+1: